Modules/RosterManager/Roster.lua

```diff
@@ -335,6 +406,7 @@ function RosterConfiguration:Set(option, value)
     if self._[option] ~= nil then
         if self:Validate(option, value) then
             self._[option] = TRANSFORMS[option](value)
+            self:PostProcess(option)
```

Comment:
This is an ugly way to solve the issue.
Why not create `SetHardCap` and `SetWeeklyCap` setters and do the processing while updating the value?

Alternatively, if `hasWeeklyCap` and `hasHardCap` are needed only for reading, why not just use a getter?

Reply:
1) This would require rewriting whole getter / setter and duplicating lots of code. 

2) because that is checked a lot of times we want to set it only once when its changed and not compare (value > 0) everytime its accessed

Reply:
> because that is checked a lot of times we want to set it only once when its changed and not compare (value > 0) everytime its accessed

This is really a non optimization. Introducing these kind of long lived cache fields has no benefit and because of the way it's coded they will go out of date at some point.

For example, I found the `Storage` api...
```lua
RosterConfiguration:Storage().weeklyCap = 17
```
Will not call your magic setter, this is why non-public stuff needs to be non-public; and in this case it isn't.

So now you have an intentionally public API allowing me to write variables and bypass both your transforms as well as your derived cached value.

Reply:
I will create a followup issue to refactor the code after we do `v0.8.0 beta` and before we go into `v1.0.0`